### DISCUSSION TITLE
Sending local image

### DISCUSSION THREAD

**Discussion by disco-research2** [bwmarrin/discordgo#1438] [Q&A]
My goal is to generate a plot from data and send the final image in the channel where the slash command was executed. So far everything works, except I'm unable to send the image.

I first registered the slash command, this works flawlessly but then when I want to execute it, it times out every time. Unfortunately, I couldn't find any similar example.
``` golang
"b-field": func(s *discordgo.Session, i *discordgo.InteractionCreate) {
	s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
		Type: discordgo.InteractionResponseChannelMessageWithSource,
		Data: &discordgo.InteractionResponseData{
			Content: "Here's the local B-field of my room:",
			Embeds: []*discordgo.MessageEmbed{
				{
					Image: &discordgo.MessageEmbedImage{URL: "PATH/plot.png"},
				},
			},
		},
	})
},
```

I'm not sure if embeds are the right approach here, but I couldn't find a better one. Btw, when entering a URL to an existing image uploaded on the internet, it works without complaining. I would tremendously appreciate some help.

**Comment by Aldiwildan77** [comment]
Local image path doesn't work in any situation, you may have to upload the file on the internet and set the URL accessible publicly.

**Reply by disco-research2** [reply]
Isn't there really no other way? It's hard to imagine that such a comprehensive package as discordgo doesn't provide any opportunity to attach local images to a message.

**Comment by disco-research2** [comment]
Found a solution by changing the function as follows:
Although I would like to have it as an embed, this works at least.

``` golang
"send_image": func(s *discordgo.Session, i *discordgo.InteractionCreate) { // <--- Stop using Single Letter Variable Names >:(
			fmt.Println("send_image.go: send_image() called")
			fileName := "PATHTOFILE"
			f, err := os.Open(fileName)
			defer f.Close()
			s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{

				Type: discordgo.InteractionResponseChannelMessageWithSource,

				Data: &discordgo.InteractionResponseData{
					Content: "Here's the local B-field of my room:",
					Files: []*discordgo.File{
						&discordgo.File{
							Name:   fileName,
							Reader: f,
						},
					},
				},
			})
		},
```

**Reply by FedorLap2006** [reply]
You can use `attachment://filename` (where `filename` is the `Name` you provided in the `File` struct) if you want to reference it in embeds.
